### PR TITLE
Cleanup cimports

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/_libs/tslib.pxd [hunk 1]
```diff
@@ -1,3 +0,0 @@
-from numpy cimport ndarray, int64_t
-
```
Comment (jreback) [inline pandas/_libs/tslib.pxd]: used in setup.py anywhere?

Comment (jbrockmendel) [reply pandas/_libs/tslib.pxd]: Nope

File: pandas/_libs/src/inference.pyx [hunk 2]
```diff
@@ -3,7 +3,7 @@ from decimal import Decimal
 cimport util
 cimport cython
 from tslibs.nattype import NaT
-from tslib cimport convert_to_tsobject
+from tslibs.conversion cimport convert_to_tsobject
```
Comment (jreback) [inline pandas/_libs/src/inference.pyx]: need to add ``conversion.pxd`` as a dep for lib.pyx (tslib.pyd should have been there)

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jbrockmendel** [comment]
ping

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**